### DISCUSSION TITLE
Where to initialize oracle database pool?

### DISCUSSION THREAD

**Discussion by csdal** [vercel/next.js#32290] [Help]
Normally in Express Node, we can initialize our connection pool in the starting `main` point of the program: `index.js`.

```
import oracledb from "oracledb";

// Initialise Database Pool
export async function initialize() {
    const dbConfig = { .. .};
    try {
        const result = await oracledb.createPool(dbConfig);
        console.log('Oracle database pool created');
        if (result) return true;
    } catch (error) {
        // console.error('Error at initializing oracle database pool: ', error);
        return false;
    }
}

// Close Database Pool
export async function close() {
    await oracledb.getPool().close(0);
}

initialize();
```

In NextJS, I find two options: 

1. It is put in `_middleware.js` and created connection pool, check pool already exists before creating.
2. It is put in custom `server.js`.

But,

 - The first one is [not able to be achieved](https://github.com/vercel/next.js/issues/32286#issuecomment-988734039) as middleware is run by [Edge Runtime](https://nextjs.org/docs/api-reference/edge-runtime).  
 - The problem with the second option is that I don't want to do and reconfigure many things in custom `server.js`. I just need database pool initialization.

My question is: Where should I keep the code of connection pool initialization?

**Comment by erdenemunkh** [comment]
Hi, Do you found any solution.

**Comment by karlguillotte** [comment]
@csdal Have you found a solution? 

More speciffically, I am experiencing same issue you were having overthere https://github.com/vercel/next.js/issues/32286 and wondering if you found a solution for the issue (node-oracledb 5.5.0 requires Node.js 8.16 or later). 

Thanks!

**Reply by karlguillotte** [reply]
Found a solution to my problem. The `oracledb` was loaded on the client. I had to reorganized my imports.

**Reply by DhruvParikh1** [reply]
Hi, what exactly was the solution? I am having the same exact issue. Thank you !

**Reply by karlguillotte** [reply]
Reorganizing my imports. I traced them down to discover they were loading on the client instead of the server.
